includes/Tracking.php

```diff
@@ -172,7 +172,7 @@ public function get_settings(): array {
 	 * @return array User properties.
 	 */
 	private function get_user_properties(): array {
-		$role        = ! empty( wp_get_current_user()->roles ) ? wp_get_current_user()->roles[0] : '';
+		$role        = ! empty( wp_get_current_user()->roles ) && \array_key_exists( 0, wp_get_current_user()->roles ) ? wp_get_current_user()->roles[0] : ''; 
```

Comment:
```suggestion
		$role        = ! empty( wp_get_current_user()->roles ) && is_array( wp_get_current_user()->roles ) ? array_shift( wp_get_current_user()->roles ) : ''; 
```

I prefer to use an array_shift here, over using the first element in an array. This is a little cleaner as well.